File: advanced-programming-methods/labs/toy-interpreter-java/src/model/expression/ConstExpression.java
package model.expression;

import exceptions.ExpressionException;
import model.IDictionary;
import model.IHeap;

/**
 * The type Const expression.
 */
public class ConstExpression extends Expression
{
    private int value;

    /**
     * Instantiates a new Const expression.
     *
     * @param value the value
     */
    public ConstExpression(int value)
    {
        this.value = value;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public int eval(IDictionary table, IHeap heap) throws ExpressionException
    {
        return this.value;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String toString()
    {
        return Integer.toString(value);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public Expression cloneDeep()
    {
        return new ConstExpression(value);
    }
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/expression/ConstExpression.cs
using System;

namespace Model
{
    /// <summary>
    /// The type Const expression.
    /// </summary>
    [Serializable()]
    public class ConstExpression : Expression
    {
        private readonly int value;

        /// <summary>
        /// Instantiates a new Const expression.
        /// </summary>
        /// <param name="value"> the value </param>
        public ConstExpression(int value)
        {
            this.value = value;
        }

        /// <inheritdoc />
        public override int Eval(IDictionary table, IHeap heap)
        {
            return this.value;
        }

        /// <summary> {@inheritDoc} </summary>
        public override string ToString()
        {
            return Convert.ToString(value);
        }

        /// <summary> {@inheritDoc} </summary>
        public override Expression CloneDeep()
        {
            return new ConstExpression(value);
        }
    }

}
